duecredit/entries.py

```diff
@@ -8,51 +8,55 @@
 #
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 import re
+from typing import Optional
 
 import logging
 lgr = logging.getLogger('duecredit.entries')
 
 
 class DueCreditEntry:
-    def __init__(self, rawentry, key=None):
+    _key : str
+
+    def __init__(self, rawentry:str, key: Optional[str] = None) -> None:
```

Comment:
note, including myself, that https://docs.python.org/3/library/typing.html#typing.Optional 

> Changed in version 3.10: Optional can now be written as `X | None`.

Reply:
I know this, but I don't know what the python baseline is for this project.

Reply:
so I see from the tests that the baseline is 3.8.

Reply:
yeah, let's stay compatible with minimal Python version ATM.